addons/mass_mailing/models/mailing.py

```diff
@@ -1048,7 +1052,12 @@ def _get_view_url(self, email_to, res_id):
         return url
 
     def action_send_mail(self, res_ids=None):
-        author_id = self.env.user.partner_id.id
+        current_partner_id = self.env.user.partner_id.id
+        if self.env['ir.model.data']._xmlid_to_res_id('base.partner_root') == current_partner_id:
```

Comment:
Actually, shouldn't we always consider the author is the "context user" computed here below (aka: mailing responsible or last writer or current user) ? Seems we could align everything.